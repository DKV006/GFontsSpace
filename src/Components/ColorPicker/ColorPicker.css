.ColorPicker .ColorPicker__Popover {
  position: absolute;
  z-index: 2;
  right: 20px; }
.ColorPicker .ColorPicker__Swatch {
  padding: 5px;
  background: #fff;
  border-radius: 1px;
  border-radius: 4px;
  height: 36px;
  width: 45px;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1);
  display: inline-block;
  cursor: pointer; }
.ColorPicker .ColorPicker__Cover {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0; }
.ColorPicker .ColorPicker__Color {
  width: 100%;
  height: 100%;
  border-radius: 2px; }

/*# sourceMappingURL=ColorPicker.css.map */
