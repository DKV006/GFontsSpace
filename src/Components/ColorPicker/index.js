import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SketchPicker } from 'react-color';
import enhanceWithClickOutside from 'react-click-outside';
import './ColorPicker.css';

class ColorPicker extends Component {
    constructor(props) {
        super(props);
        this.state = {
            color: this.props.color,
            displayColorPicker: false
        }
        this.handOnClick = this.handOnClick.bind(this);
        this.handOnChange = this.handOnChange.bind(this);
        this.handOnClose = this.handOnClose.bind(this);
    }

    handOnClick() {
        this.setState({ displayColorPicker: !this.state.displayColorPicker })
    }

    handOnChange(color) {
        this.setState({ color: color.rgb },()=>{
            if(this.props.isBackgroundColorPicker )
            document.body.style.backgroundColor = `rgba(${this.state.color.r}, ${this.state.color.g}, ${this.state.color.b}, ${this.state.color.a})`;
        })
        // this.props.handOnChange()
    }

    handleClickOutside() {
        this.setState({ displayColorPicker: false })
    }

    handOnClose() {
    }

    __getStyleObject(isBackgroundColorPicker) {
        const colorObject = {
            color: `rgba(${this.state.color.r}, ${this.state.color.g}, ${this.state.color.b}, ${this.state.color.a})`,
        };

        const backGroundObject = {
            backgroundColor: `rgba(${this.state.color.r}, ${this.state.color.g}, ${this.state.color.b}, ${this.state.color.a})`,
        };

        return isBackgroundColorPicker ? backGroundObject : colorObject

    }

    render() {
        return (
            <div className="ColorPicker">
                <div className="ColorPicker__Swatch" onClick={this.handOnClick}>
                    <div className="ColorPicker__Color" style={this.__getStyleObject(this.props.isBackgroundColorPicker)}>
                        {!this.props.isBackgroundColorPicker && "A"}
                    </div>
                </div>
                {this.state.displayColorPicker ?
                    <div className="ColorPicker__Popover">
                        <SketchPicker color={this.state.color} onChange={this.handOnChange} />
                    </div> :
                    null}

            </div>
        );
    }
}

ColorPicker.propTypes = {

};

export default enhanceWithClickOutside(ColorPicker);