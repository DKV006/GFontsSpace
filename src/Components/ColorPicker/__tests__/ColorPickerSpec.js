import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import ColorPicker from './../index';

describe('ColorPicker', () => {
    const colorPickerProps = {
        r: '0',
        g: '0',
        b: '0',
        a: '1',
    }
    it('renders correctly', () => {
        let tree = renderer.create(
            <ColorPicker color={colorPickerProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
