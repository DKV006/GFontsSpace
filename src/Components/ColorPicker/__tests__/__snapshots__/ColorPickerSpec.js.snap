// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders correctly 1`] = `
<div
  className="ColorPicker"
>
  <div
    className="ColorPicker__Swatch"
    onClick={[Function]}
  >
    <div
      className="ColorPicker__Color"
      style={
        Object {
          "color": "rgba(0, 0, 0, 1)",
        }
      }
    >
      A
    </div>
  </div>
</div>
`;
